### PR TITLE
Fix for SendGrid Batches API 

### PR DESCRIPTION
**Pull request by spib**
OK, this is stupid, but it looks like the SendGrid Batches API (https://sendgrid.com/docs/API_Reference/Web_API_v3/cancel_schedule_send.html) returns an array of results when you request the status of a single batch. StrongGrid (sensibly) expects it to be a single item.

This PR fixes the bug

### CODE REVIEW

**Comment by Jericho** [comment]
Unbelievable! We request the status of **one** batch and we receive an array?!?!?! Thanks for discovering this. I agree, of course, that we need to fix it but I think the signature of `IBatches.GetAsync` should remain same and return a single `BatchInfo`. Developers expect a single item when they request the status of a single batch!

**Comment by Jericho** [comment]
By the way, thanks for providing unit tests. Much appreciated!

**Comment by spib** [comment]
No problem, feel free to change the fix to make the signature consistent

File: Source/StrongGrid/Resources/Batches.cs [hunk 1]
```diff
@@ -135,12 +135,12 @@ public Task<BatchInfo[]> GetAllAsync(CancellationToken cancellationToken = defau
...
 		/// A <see cref="BatchInfo" />.
 		/// </returns>
-		public Task<BatchInfo> GetAsync(string batchId, CancellationToken cancellationToken = default(CancellationToken))
+		public Task<BatchInfo[]> GetAsync(string batchId, CancellationToken cancellationToken = default(CancellationToken))
 		{
 			return _client
 				.GetAsync($"user/scheduled_sends/{batchId}")
 				.WithCancellationToken(cancellationToken)
-				.AsSendGridObject<BatchInfo>();
+				.AsSendGridObject<BatchInfo[]>();
```
[the remark this answers is not in the document]
Comment (spib) [reply Source/StrongGrid/Resources/Batches.cs]: How do you return FirstOrDefault() on a Task<BatchInfo[]>?

[the remark this answers is not in the document]
Comment (Jericho) [reply Source/StrongGrid/Resources/Batches.cs]: I would do something like this:

```
public async Task<BatchInfo> GetAsync(string batchId, CancellationToken cancellationToken = default(CancellationToken))
{
	var result = await _client
		.GetAsync($"user/scheduled_sends/{batchId}")
		.WithCancellationToken(cancellationToken)
		.AsSendGridObject<BatchInfo[]>()
		.ConfigureAwait(false);

	// For some reason, SendGrid returns an array containing a single
	// item when we requested the status for a specific batch
	return result.FirstOrDefault();
}
```

**Comment by spib** [comment]
Should be good now

**Comment by spib** [comment]
Great - when do you expect the release with this fix to be available? It'd be great to get it sooner rather than later

**Comment by Jericho** [comment]
I was hopping to get an answer to this [question](https://stackoverflow.com/questions/46508589/how-to-calculate-the-size-of-a-httprequestmessage) which would allow me to add a new feature where I make sure the size of the email the developer is trying to send does not exceeds the max size allowed by SendGrid but I guess this new feature will have to wait. 

Working on another project at the moment but I'll release 0.30.0 ASAP. Either later today or tomorrow.

**Comment by Jericho** [comment]
Done: 0.30.0 published to nuget

**Pull request merged**